addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ApplicationErrorScanRuleUnitTest.java

```diff
@@ -347,6 +350,101 @@ void shouldNotRaiseAlertForResponseCodeOkAndContentTypeWebAssemblyWhenFilePayloa
         assertThat(alertsRaised.size(), equalTo(0));
     }
 
+    private void createHttpMessageWithContentTypeAndCustomEvidence(
+            HttpMessage msg, String contentType, String errorString)
+            throws HttpMalformedHeaderException {
+        msg.setRequestHeader(REQUEST_HEADER);
+        msg.setResponseBody(createResponseHeader(OK));
+        msg.getResponseHeader().addHeader(HttpHeader.CONTENT_TYPE, "text/" + contentType);
+        msg.setResponseBody(errorString);
+    }
+
+    private void createHttpMessageWithUriFilePathAndCustomEvidence(
+            HttpMessage msg, String contentType, String errorString)
+            throws HttpMalformedHeaderException {
+        String filePath = "";
+        if (contentType.equals("javascript")) {
+            filePath = "/test.js";
+        } else if (contentType.equals("css")) {
+            filePath = "/test.css";
+        }
+        msg.setRequestHeader("GET https://www.example.com/" + filePath + " HTTP/1.1");
+        msg.setResponseBody(createResponseHeader(OK));
+        msg.setResponseBody(errorString);
+    }
+
+    @ParameterizedTest
+    @MethodSource("contentTypeProvider")
+    void
+            shouldNotRaiseAlertForResponseCodeOkAndContentTypeJavaScriptOrCssWithFilePayloadPresentWhenThresholdNotLow(
+                    String contentType, String isUriOrResponseBody)
+                    throws HttpMalformedHeaderException {
+        // Given
+        // String from standard XML file
+        String errorString = "Microsoft OLE DB Provider for ODBC Drivers";
+        HttpMessage msg = new HttpMessage();
+        if (isUriOrResponseBody.equals("uri"))
+            createHttpMessageWithUriFilePathAndCustomEvidence(msg, contentType, errorString);
+        else if (isUriOrResponseBody.equals("responseBody"))
+            createHttpMessageWithContentTypeAndCustomEvidence(msg, contentType, errorString);
```

Comment:
I'm referring to this duplication, this should be done by the source instead of duplicating in each method.